### ISSUE TITLE
False positive for testing-library/no-await-sync-events

### ISSUE THREAD

**Issue by nickmccurdy** [testing-library/eslint-plugin-testing-library#403]
Using `await` with `userEvent.type`'s `delay` option is correct, but is incorrectly flagged as a lint error.

```js
async () => {
  const delay = 10
  // error  `userEvent.type` is sync and does not need `await` operator  testing-library/no-await-sync-events
  await userEvent.type(element, text, {delay})
}
  ```

**Comment by nickmccurdy** [comment]
> Is the plugin not picking up the overload anymore ? :disappointed: 

This might be just another reason for us to always return a Promise on `userEvent.type` in the next major.
@ph-fritsche in https://github.com/testing-library/user-event/pull/693#discussion_r664250440
